### PR TITLE
refactor(lodash): remove noop usage

### PR DESCRIPTION
**Pull request by francoischalifour**
This removes `lodash/noop` for a custom `noop` util function.

### CODE REVIEW

**Comment by Haroenv** [comment]
Do you think we should enforce this? Maybe possible with eslint

File: src/lib/utils/noop.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function noop() {}
+
+export default noop;
```
Comment (tkrugg) [inline src/lib/utils/noop.ts]: typing wise, do you knwo why they have `noop(...args: any[]): void;` in lodash and if there is any benefit to it?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/06d8677cde28467e830bf77bc2cbfdaca37740aa/types/lodash/common/util.d.ts#L885

Comment (Haroenv) [reply src/lib/utils/noop.ts]: this means that you can call a noop without issues; I think we might not allow our noop to be called

Comment (francoischalifour) [reply src/lib/utils/noop.ts]: I guess it's to be more flexible. I'll update that.

Comment (francoischalifour) [reply src/lib/utils/noop.ts]: Hmm I didn't see your comment @Haroenv. I updated the signature.

File: src/lib/InstantSearch.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import simpleMapping from './stateMappings/simple';
 import historyRouter from './routers/history';
```
Comment (samouss) [inline src/lib/InstantSearch.js]: Inside the file we have a line with:

```js
this._onHistoryChange = function() {};
```

We could also replace it with `noop`.

File: src/lib/utils/__tests__/noop-test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import noop from '../noop';
+
+describe('noop', () => {
+  test('does nothing', () => {
+    expect(noop).toBeInstanceOf(Function);
+    expect(noop()).toBeUndefined();
+    // Checks that `noop` acts like a function.
+    expect(noop.toString).toBe(Function.prototype.toString);
+  });
+});
```
Comment (tkrugg) [inline src/lib/utils/__tests__/noop-test.js]: it's nitpicking but this file can be typescript

**Pull request merged**